core/src/main/scala/org/apache/spark/rdd/PipedRDD.scala

```diff
@@ -133,23 +136,30 @@ private[spark] class PipedRDD[T: ClassTag](
     // Start a thread to feed the process input from our parent's iterator
     new Thread("stdin writer for " + command) {
       override def run() {
-        TaskContext.setTaskContext(context)
         val out = new PrintWriter(proc.getOutputStream)
 
-        // scalastyle:off println
-        // input the pipe context firstly
-        if (printPipeContext != null) {
-          printPipeContext(out.println(_))
-        }
-        for (elem <- firstParent[T].iterator(split, context)) {
-          if (printRDDElement != null) {
-            printRDDElement(elem, out.println(_))
-          } else {
-            out.println(elem)
+        try {
+          TaskContext.setTaskContext(context)
```

Comment:
It won't matter for correctness, but does mean the ordering of these changes. I can't see how it would make a difference here anyway since these should be unrelated, so it's not a problem. I'd say the opposite though: you want to restrict this try-finally block, which really exists to manage the life of `out`, to statements that relate to use `out`.